### ISSUE TITLE
No categories in Standard CEs

### ISSUE THREAD

**Issue by bernhardberger** [TYPO3-Headless/headless#689]
**Describe the bug**

**To Reproduce**
Create a CE (doesn't seem to matter which, tried Images and File Links) and add Categories
![image](https://github.com/TYPO3-Headless/headless/assets/2110083/d833ff82-9ee0-4a47-84c9-a29e9fd76538)

API doesn't output any categories, just an empty string.
![image](https://github.com/TYPO3-Headless/headless/assets/2110083/f3e151b6-e2c8-4ac8-825b-3c265325b8cf)

**Expected behavior**
To get Categories in any form

**TYPO3 version and TYPO3 Headless version**
`TYPO3 12.4.8`, `EXT:headless 4.2.3`

**Comment by schloram** [comment]
@bernhardberger Where are your category records stored?

**Comment by bernhardberger** [comment]
**Not** on `pid=0`.

I figured it out that the TS select only gets categories from the root pid. Not sure if that's a good default tbh, especially for sites with multi tree.

**Comment by schloram** [comment]
@bernhardberger Yes that is he issue here. Even though the [TYPO3 Documentation](https://docs.typo3.org/m/typo3/reference-typoscript/main/en-us/Functions/Select.html#t3-function-select-pidInList) has an example to get categories from `root` I don't think it's ideal since you might want to have different categories for different site roots.

In this case it would be better to use `pidInList = leveluid : 0` with `recursive` imho. But since this would ignore the records from `root` (page uid 0) completely you would be better of overriding `lib.contentElement.fields.categories` with these changes yourself.

Another approach would be using `pidInList = root` with `recursive` to get every category from the `root` and below.

**Comment by lukaszuznanski** [comment]
I think it needs to be specified per project, but we can think of better default setting. @schloram approach with `pidInList = root` and `recursive` sounds good for starting point.
